### PR TITLE
Tyler lundgren 1801 final

### PR DESCRIPTION
**Pull request by nergdnvlt**
Think it felt successful, predicated of course on if I read the user stories correctly.

### CODE REVIEW

**Comment by nergdnvlt** [comment]
A head's up: for some reason, I didn't have users in the repo I pulled, so I had to build that out. Talking to other people, I guess it was already supposed to be there?

File: spec/features/genres/visitor_can_see_genres_but_not_form_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+describe "As an admin" do
```
Comment (AliSchlereth) [inline spec/features/genres/visitor_can_see_genres_but_not_form_spec.rb]: Your test file says as a visitor, but this description section says as an admin.

File: spec/features/genres/visitor_can_see_genres_but_not_form_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      assert has_no_field?(:name)
+
+      expect(page).to have_content(genre1.name)
+      expect(page).to have_content(genre2.name)
+      expect(page).to have_content(genre3.name)
+    end
+  end
+
+  context "they visit the genre index" do
+    it "and fill in a form to create a new genre" do
```
Comment (AliSchlereth) [inline spec/features/genres/visitor_can_see_genres_but_not_form_spec.rb]: This it block says the user is going to fill in the form, but that isn't what this test is about.

File: spec/features/genres/admin_can_visit_genre_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    it "and fill in a form to create a new genre" do
+      admin = User.create!(username: 'fluffy', password: 'test', role: 'admin')
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(admin)
+      visit genres_path
+
+      fill_in 'genre[name]', with: 'Action'
+      click_on 'Create Genre'
+
+      expect(current_path).to eq(genres_path)
+      expect(page).to have_content(Genre.all.last.name)
```
[the remark this answers is not in the document]
Comment (AliSchlereth) [reply spec/features/genres/admin_can_visit_genre_index_spec.rb]: I'd also want to see a sad path test showing what would happen when a genre is not successfully created.

File: app/controllers/genres_controller.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  def create
+    @genre = Genre.new(genre_params)
+    if @genre.save
+      flash[:success] = "#{@genre.name} Created!"
+    else
+      flash[:error] = "Genre Not Created!"
+    end
+    redirect_to genres_path
+  end
```
Comment (AliSchlereth) [inline app/controllers/genres_controller.rb]: Only admin should be able to create a genre. How can you ensure this behavior?

**Comment by AliSchlereth** [comment]
@nergdnvlt  This is pretty solid. There are a few things we'd like you to clean up:
* Scope genre create to admin only
* I like how you built a communicative sad path for when genres don't successfully create, but you are missing a test for this piece of functionality
* Many of you blocks have the same description in your describe/it blocks in your tests

**Comment by nergdnvlt** [comment]
@alischlereth Do you want action on these revisions? I can namespace the create action and provide a failure test pretty quickly as soon as I’m off the bus - or is this just recommendations for the future? 

Also - totally my bad on not paying attention to my test names. No excuse. 🤦🏽‍♂️

**Pull request closed without merging**